ksqldb-rest-app/src/test/java/io/confluent/ksql/api/integration/RestoreCommandTopicMultipleKafkasIntegrationTest.java

```diff
@@ -0,0 +1,292 @@
+        .withProperty(KsqlRestConfig.COMMAND_PRODUCER_PREFIX + StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, INTERNAL_TEST_HARNESS.kafkaBootstrapServers())
+        .build();
+  }
+
+  @Before
+  public void setup() throws IOException {
+    REST_APP.start();
+
+    final KsqlConfig ksqlConfig = new KsqlConfig(REST_APP.getKsqlRestConfig().getKsqlConfigProperties());
+    commandTopic = ReservedInternalTopics.commandTopic(ksqlConfig);
+    backupFile = Files.list(BACKUP_LOCATION.toPath()).findFirst().get();
+    propertiesFile = TMP_FOLDER.newFile().toPath();
+    writeServerProperties(propertiesFile);
+  }
+
+  @After
+  public void teardown() {
+    REST_APP.stop();
+    INTERNAL_TEST_HARNESS.deleteTopics(Collections.singletonList(commandTopic));
+  }
+
+  @After
+  public void teardownClass() {
+    TMP_FOLDER.delete();
+  }
+
+  private static void writeServerProperties(final Path propertiesFile) throws IOException {
+    final Map<String, Object> map = REST_APP.getKsqlRestConfig().getKsqlConfigProperties();
+
+    Files.write(
+            propertiesFile,
+        map.keySet().stream()
+            .map((key -> key + "=" + map.get(key)))
+            .collect(Collectors.joining("\n"))
+            .getBytes(StandardCharsets.UTF_8),
+        StandardOpenOption.CREATE
+    );
+  }
+
+  @Test
+  public void shouldBackupAndRestoreCommandTopic() throws Exception {
+    // Given
+    TEST_HARNESS.ensureTopics("topic1", "topic2");
+
+    makeKsqlRequest("CREATE STREAM TOPIC1 (ID INT) "
+        + "WITH (KAFKA_TOPIC='topic1', VALUE_FORMAT='JSON');");
+    makeKsqlRequest("CREATE STREAM TOPIC2 (ID INT) "
+        + "WITH (KAFKA_TOPIC='topic2', VALUE_FORMAT='JSON');");
+    makeKsqlRequest("CREATE STREAM stream1 AS SELECT * FROM topic1;");
+    makeKsqlRequest("CREATE STREAM stream2 AS SELECT * FROM topic2;");
+
+    // When
+
+    // Delete the command topic and check the server is in degraded state
+    INTERNAL_TEST_HARNESS.deleteTopics(Collections.singletonList(commandTopic));
+    assertThatEventually("Degraded State", this::isDegradedState, is(true));
+
+    // Restore the command topic
+    KsqlRestoreCommandTopic.main(
+        new String[]{
+            "--yes",
+            "--config-file", propertiesFile.toString(),
+            backupFile.toString()
+        });
+
+    // Re-load the command topic
+    REST_APP.stop();
+    REST_APP.start();
+
+    // Then
+    final List<String> streamsNames = showStreams();
+    assertThat("Should have TOPIC1", streamsNames.contains("TOPIC1"), is(true));
+    assertThat("Should have TOPIC2", streamsNames.contains("TOPIC2"), is(true));
+    assertThat("Should have STREAM1", streamsNames.contains("STREAM1"), is(true));
+    assertThat("Should have STREAM2", streamsNames.contains("STREAM2"), is(true));
+    assertThat("Server should NOT be in degraded state", isDegradedState(), is(false));
+  }
+
+  @Test
+  public void shouldSkipIncompatibleCommands() throws Exception {
```

Comment:
I think it's alright to leave out this test, we're mainly just testing the two broker setup.